### ISSUE TITLE
Support expand of types

### ISSUE THREAD

**Issue by campoy** [dgraph-io/dgraph#3903]
Since v1.1.0 Dgraph support types and `expand(_all_)` is defined as the expansion of all the predicates found in all of the types linked to any of the nodes in the query.

So for instance, if we have two types Person and Aged:

```graphql
type Person {
   first_name: string
   last_name: string
}

type Aged {
   age: int
}
```

And we have the following dataset:

```
{
  set {
    _:a <dgraph.type> "Person" .
    _:a <dgraph.type> "Aged" .
    _:a <first_name> "John" .
    _:a <last_name> "Doe" .
    _:a <age> "42" .
     }
}
```

When we query all values of type `Person` and use `expand(_all_)`:

```graphql
{
  q(func: type(Person)) {
    dgraph.type
    expand(_all_)
  }
}
```

we end up receiving the field `age` even though it's not part of the type `Person` because `expand(_all_)` expands all of the predicates in all of the types linked to any of the values returned.

```graphql
{
  "data": {
    "q": [
      {
        "dgraph.type": [
          "Person",
          "Aged"
        ],
        "first_name": "John",
        "age": 42,
        "last_name": "Doe"
      }
    ]
  },
  "extensions": {...}
}
```

This is because our node `John Doe` is of type `Person` and `Aged`, as the result of the query shows.

I propose a way to expand only the fields related to a type by passing the type name as the parameter to `expand`.

So, we could send this query:

```graphql
{
  q(func: type(Person)) {
    dgraph.type
    expand(Person)
  }
}
```

This would return only the fields `first_name` and `last_name`. We could also pass multiple types to the expand query: `expand(Person, Aged)` and that would expand the union of the predicates.

**Comment by martinmr** [comment]
@campoy How is this different from #3904 It looks like similar functionality but with different syntax.

**Comment by martinmr** [comment]
Fixed by #3920
